python/lsst/meas/modelfit/psf.py

```diff
@@ -105,18 +104,35 @@ def measure(self, measRecord, exposure):
         psf = exposure.getPsf()
         psfImage = psf.computeKernelImage(measRecord.getCentroid())
         psfShape = psf.computeShape(measRecord.getCentroid())
+        lastError = None
+        lastModel = None
         # Fit the first element in the sequence, using the PSFs moments to initialize the parameters
-        fitter, key = self.sequence[0]
-        lastResult = fitter.apply(psfImage, psfShape)
-        lastModel = fitter.getModel()
-        measRecord.set(key, lastResult)
         # For every other element in the fitting sequence, use the previous fit to initialize the parameters
-        for fitter, key in self.sequence[1:]:
-            initial = fitter.adapt(lastResult, lastModel)
-            lastResult = fitter.apply(psfImage, initial)
-            lastModel = fitter.getModel()
-            measRecord.set(key, lastResult)
-
+        for fitter, name in self.sequence:
+            try:
+                if lastModel is None:
+                    fitter.measure(measRecord, psfImage, psfShape)
+                else:
+                    fitter.measure(measRecord, psfImage, fitter.adapt(lastResult, lastModel))
+                lastResult = measRecord.get(fitter.getKey())
+                lastModel = fitter.getModel()
+            except lsst.meas.base.baseMeasurement.FATAL_EXCEPTIONS:
+                raise
+            except lsst.meas.base.baseLib.MeasurementError as error:
+                fitter.fail(measRecord, error.cpp)
+                lastError = error
+            except Exception as error:
+                fitter.fail(measRecord)
+                lastError = error
+        # When we are done with all the fitters, raise the last error if there was one.
+        # This gives the calling task a chance to do whatever it wants
+        if not lastError is None:
+            raise(lastError)
```

Comment:
I think the syntax you want here is just `raise lastError`.  The parenthesis don't do any harm (they're just doing grouping), but they should still be removed to avoid confusion.